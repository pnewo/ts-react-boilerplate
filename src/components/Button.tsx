import * as React from 'react';

export interface IButtonProps {
    click(): void;
    readonly text: string;
}

const Button: React.StatelessComponent<IButtonProps> = ({ click, text }) => (
    <input className="btn" type="submit" onClick={() => click()} value={text} />
);

export default Button;
