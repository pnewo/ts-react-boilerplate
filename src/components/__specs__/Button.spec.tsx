import * as React from 'react';
import { shallow } from 'enzyme';
import Button from '../Button';

describe('Button', () => {
    const click = jest.fn();
    const wrapper = shallow(<Button click={click} text="" />);

    it('should render correctly', () => (
        expect(wrapper).toMatchSnapshot()
    ));

    it('should call the correct function on click', () => {
        const testTimes = 5;
        for (let i = 0; i < testTimes; i++) {
            wrapper.find('.btn').simulate('click');
        }
        expect(click).toHaveBeenCalledTimes(testTimes); 
    });
});
