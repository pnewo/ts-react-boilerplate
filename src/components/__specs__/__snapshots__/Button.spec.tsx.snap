// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<input
  className="btn"
  onClick={[Function]}
  type="submit"
  value=""
/>
`;
